bundles/org.openhab.binding.smarther/src/main/java/org/openhab/binding/smarther/internal/account/SmartherAccountService.java

```diff
@@ -0,0 +1,295 @@
+        if (sender != null) {
+            final SmartherAccountHandler listener = getAccountListenerByLocation(sender.getPlant().getId());
+            if (listener == null) {
+                logger.warn("C2C notification [{}]: no matching bridge was found. Possible bridge has been removed.",
+                        notification.getId());
+                throw new SmartherGatewayException(ERROR_UKNOWN_BRIDGE);
+            } else if (listener.isOnline()) {
+                final SmartherNotificationHandler handler = (SmartherNotificationHandler) listener;
+
+                if (handler.useNotifications()) {
+                    // Passes the notification to the handler
+                    handler.handleNotification(notification);
+                } else {
+                    logger.debug(
+                            "C2C notification [{}]: notification discarded as bridge does not handle notifications.",
+                            notification.getId());
+                }
+            } else {
+                logger.debug("C2C notification [{}]: notification discarded as bridge is offline.",
+                        notification.getId());
+            }
+        } else {
+            logger.debug("C2C notification [{}]: notification discarded as payload is invalid.", notification.getId());
+        }
+    }
+
+    /**
+     * Adds a {@link SmartherAccountHandler} handler to the set of account service handlers.
+     *
+     * @param handler
+     *            the handler to add to the handlers set
+     */
+    public void addSmartherAccountHandler(SmartherAccountHandler handler) {
+        if (!handlers.contains(handler)) {
+            handlers.add(handler);
+        }
+    }
+
+    /**
+     * Removes a {@link SmartherAccountHandler} handler from the set of account service handlers.
+     *
+     * @param handler
+     *            the handler to remove from the handlers set
+     */
+    public void removeSmartherAccountHandler(SmartherAccountHandler handler) {
+        handlers.remove(handler);
+    }
+
+    /**
+     * Returns all the {@link SmartherAccountHandler} account service handlers.
+     *
+     * @return a set containing all the account service handlers
+     */
+    public Set<SmartherAccountHandler> getSmartherAccountHandlers() {
+        return handlers;
+    }
+
+    /**
+     * Searches the {@link SmartherAccountHandler} handler that matches the given Thing UID.
+     *
+     * @param thingUID
+     *            the UID of the Thing to match the handler with
+     *
+     * @return the handler matching the given Thing UID, or {@code null} if none matches
+     */
+    private @Nullable SmartherAccountHandler getAccountListenerByUID(String thingUID) {
+        final Optional<SmartherAccountHandler> maybeListener = handlers.stream().filter(l -> l.equalsThingUID(thingUID))
+                .findFirst();
+        return (maybeListener.isPresent()) ? maybeListener.get() : null;
```

Comment:
Seems not, I receive an error that orElse argument cannot be null.
Found a strange javadoc, indeed:

```
**@NonNull SmartherAccountHandler java.util.Optional.orElse(@NonNull SmartherAccountHandler other)**

Return the value if present, otherwise return other.

Parameters:
    **other** the value to be returned if there is no value present, maybe null
```

Reply:
well it was worth a shot, just leave as is then.